frontend/src/components/pages/organization/OrgShop.tsx

```diff
@@ -1,32 +1,76 @@
 import { useQuery } from "@apollo/client";
-import { Box, Grid, Stack, Typography } from "@mui/material";
+import { Box, Grid, Stack, Typography, Button} from "@mui/material";
+import {KeyboardArrowLeft, KeyboardArrowRight} from '@mui/icons-material';
+import { useTheme } from '@mui/material/styles';
 
 import { AdminOrganizationFragment, AllShopOrdersDocument } from "@/generated/graphql";
-
 import { ShopSale } from "../orgs/ShopSale";
+import React,{ useState } from 'react';
 
 type Props = {
   organization: AdminOrganizationFragment;
 };
 export const OrgProducts: React.FC<Props> = ({ organization }) => {
-  const { data, error } = useQuery(AllShopOrdersDocument);
+
+  const limit = 5
+  const [page, setPage] = useState(0);
+
+
+  const { data, error, loading } = useQuery(AllShopOrdersDocument, {
+    variables: {
+      limit: limit + 1,    // The number of orders you want to fetch
+      offset: page * limit,    // The starting index (e.g., 0 for the first set of results)
+    },
+  });
   if (error) return <p>Error</p>;
 
-  console.log(data);
-  if (organization.name !== "Janus linjeforening") {
+  if (organization.name.toLowerCase() !== "janus linjeforening") {
     return (
       <p>
         Per nå har kun Janus tilgang på buttikk administrasjon. Etter hvert vil vi åpne for at flere kan bruke siden
       </p>
     );
   }
-  if (data?.allShopOrders?.length === 0) {
+  if (data?.allShopOrders!.length === 0) {
     return <p>Ingen ordre</p>;
   }
-  return (
-    <>
-      <Stack direction={"row"} padding={2} border={3}>
-        <Box display="flex" alignItems="left" justifyContent="left" width={"15%"} padding={1}>
+    // Pagination button handlers
+    const nextPage = () => setPage(page + 1);
+    const prevPage = () => setPage(page > 0 ? page - 1 : 0);
+    const theme = useTheme();
+    // Check if there is a next page (if we received less than `limit` items)
+    const hasNextPage = (data?.allShopOrders && data.allShopOrders.length < limit) ?? false; //Have to check if the data exists before I check that the data has a certain length. This gave me an error: data?.allShopOrders?.length < limit;
```

Comment:
The comment is not needed here. Also this is an example of why the query should be renamed. When it is called allShopOrders it seems like you have a bug here, but i dont think  you actually do because data.allShopOrders isn't actually all the shop orders but rather the ones in the current query.

Reply:
Name changed, comment removed